ona/models.py

```diff
@@ -0,0 +1,32 @@
+from django.contrib.postgres.fields import HStoreField
+from django.db import models
+from accounts.models import Integration
+from reports.models import Report
+
+
+class Submission(models.Model):
+
+    integration = models.ForeignKey(Integration,
+                                    related_name='submissions',
+                                    null=False)
+    report = models.ForeignKey(Report,
+                               related_name='submissions',
+                               null=True)
+    content = models.TextField(null=False, blank=False)
+    metadata = HStoreField(null=True, blank=True, default={})
+    created_at = models.DateTimeField(auto_now_add=True)
+    submitted = models.BooleanField(default=False)
+
+    def __str__(self):
+        return self.content
+
+# send new submissions
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from .tasks import send_submission
```

Comment:
Ah, one of Python's annoying corners. A common practice is then to do the import inside the function that uses it, e.g.:
```python
def fire_subm_action_if_undelivered(sender, instance, created, **kwargs):
    from .tasks import send_submission
    if not instance.submitted:
        send_submission.delay(str(instance.id))
```
Python caches imports so this is not too slow after the first time.